internal/api/mfa.go

```diff
@@ -282,7 +289,7 @@ func (a *API) VerifyFactor(w http.ResponseWriter, r *http.Request) error {
 				return err
 			}
 		}
-		return unprocessableEntityError(ErrorCodeMFAVerificationFailed, "Invalid TOTP code entered")
+		return unprocessableEntityError(ErrorCodeMFAVerificationFailed, "Invalid TOTP code entered: %v", params.Code).WithInternalError(verr)
```

Comment:
Let's not log full codes please. We can log a hash of the code, or the first letter/digit.

Even though the code is "invalid" an attacker with access to the logs could be able to reconstruct state on the client. For example, imagine the client's clock is lagging behind 30 seconds. An attacker could spot that from the logs, grab the code and impersonate them.